ext/mysqli/mysqli_api.c

```diff
@@ -811,15 +811,60 @@ PHP_FUNCTION(mysqli_stmt_execute)
 {
 	MY_STMT		*stmt;
 	zval		*mysql_stmt;
+	HashTable	*input_params = NULL;
 #ifndef MYSQLI_USE_MYSQLND
 	unsigned int	i;
 #endif
 
-	if (zend_parse_method_parameters(ZEND_NUM_ARGS(), getThis(), "O", &mysql_stmt, mysqli_stmt_class_entry) == FAILURE) {
+	if (zend_parse_method_parameters(ZEND_NUM_ARGS(), getThis(), "O|h", &mysql_stmt, mysqli_stmt_class_entry, &input_params) == FAILURE) {
 		RETURN_THROWS();
 	}
 	MYSQLI_FETCH_RESOURCE_STMT(stmt, mysql_stmt, MYSQLI_STATUS_VALID);
 
+	// bind-in-execute
+	if (input_params) {
```

Comment:
I think we need to make the argument `?array $params = null`, because calling `$stmt->execute()` is *not* the same as `$stmt->execute([])`. The former doesn't bind anything (requiring manual binding), the latter binds zero parameters.

cc @kocsismate